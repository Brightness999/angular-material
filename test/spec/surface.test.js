/**
 * @fileOverview Test the API surface, library properly exporting methods.
 */

const Logality = require('../..');

describe('API Surface', () => {
  test('Logality is a function', () => {
    expect(typeof Logality).toBe('function');
  });
  test('Logality instantiates properly', () => {
    const logality = new Logality();
    expect(typeof logality.get).toBe('function');
  });
  test('Logality instantiates without the "new" keyword', () => {
    const logality = Logality();
    expect(typeof logality.get).toBe('function');
  });
  test('Logality instance has expected methods exported', () => {
    const logality = Logality();

    expect(typeof logality.get).toBe('function');
    expect(typeof logality.log).toBe('function');
  });
});
