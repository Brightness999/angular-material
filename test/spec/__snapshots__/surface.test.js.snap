// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`API Surface Logging Function Levels Testing log level: alert 1`] = `
Object {
  "context": Object {
    "runtime": Object {
      "application": "testLogality",
      "file": "/test/spec/surface.test.js",
    },
    "source": Object {
      "file_name": "/test/spec/surface.test.js",
    },
    "system": Object {
      "hostname": "127.0.0.1",
      "pid": 36255,
      "process_name": "/Users/thanpolas/.nvm/versions/node/v10.1.0/bin/node",
    },
  },
  "dt": "2018-05-18T16:25:57.815Z",
  "event": Object {},
  "level": "alert",
  "message": "hello world",
}
`;

exports[`API Surface Logging Function Levels Testing log level: critical 1`] = `
Object {
  "context": Object {
    "runtime": Object {
      "application": "testLogality",
      "file": "/test/spec/surface.test.js",
    },
    "source": Object {
      "file_name": "/test/spec/surface.test.js",
    },
    "system": Object {
      "hostname": "127.0.0.1",
      "pid": 36255,
      "process_name": "/Users/thanpolas/.nvm/versions/node/v10.1.0/bin/node",
    },
  },
  "dt": "2018-05-18T16:25:57.815Z",
  "event": Object {},
  "level": "critical",
  "message": "hello world",
}
`;

exports[`API Surface Logging Function Levels Testing log level: debug 1`] = `
Object {
  "context": Object {
    "runtime": Object {
      "application": "testLogality",
      "file": "/test/spec/surface.test.js",
    },
    "source": Object {
      "file_name": "/test/spec/surface.test.js",
    },
    "system": Object {
      "hostname": "127.0.0.1",
      "pid": 36255,
      "process_name": "/Users/thanpolas/.nvm/versions/node/v10.1.0/bin/node",
    },
  },
  "dt": "2018-05-18T16:25:57.815Z",
  "event": Object {},
  "level": "debug",
  "message": "hello world",
}
`;

exports[`API Surface Logging Function Levels Testing log level: emergency 1`] = `
Object {
  "context": Object {
    "runtime": Object {
      "application": "testLogality",
      "file": "/test/spec/surface.test.js",
    },
    "source": Object {
      "file_name": "/test/spec/surface.test.js",
    },
    "system": Object {
      "hostname": "127.0.0.1",
      "pid": 36255,
      "process_name": "/Users/thanpolas/.nvm/versions/node/v10.1.0/bin/node",
    },
  },
  "dt": "2018-05-18T16:25:57.815Z",
  "event": Object {},
  "level": "emergency",
  "message": "hello world",
}
`;

exports[`API Surface Logging Function Levels Testing log level: error 1`] = `
Object {
  "context": Object {
    "runtime": Object {
      "application": "testLogality",
      "file": "/test/spec/surface.test.js",
    },
    "source": Object {
      "file_name": "/test/spec/surface.test.js",
    },
    "system": Object {
      "hostname": "127.0.0.1",
      "pid": 36255,
      "process_name": "/Users/thanpolas/.nvm/versions/node/v10.1.0/bin/node",
    },
  },
  "dt": "2018-05-18T16:25:57.815Z",
  "event": Object {},
  "level": "error",
  "message": "hello world",
}
`;

exports[`API Surface Logging Function Levels Testing log level: info 1`] = `
Object {
  "context": Object {
    "runtime": Object {
      "application": "testLogality",
      "file": "/test/spec/surface.test.js",
    },
    "source": Object {
      "file_name": "/test/spec/surface.test.js",
    },
    "system": Object {
      "hostname": "127.0.0.1",
      "pid": 36255,
      "process_name": "/Users/thanpolas/.nvm/versions/node/v10.1.0/bin/node",
    },
  },
  "dt": "2018-05-18T16:25:57.815Z",
  "event": Object {},
  "level": "info",
  "message": "hello world",
}
`;

exports[`API Surface Logging Function Levels Testing log level: notice 1`] = `
Object {
  "context": Object {
    "runtime": Object {
      "application": "testLogality",
      "file": "/test/spec/surface.test.js",
    },
    "source": Object {
      "file_name": "/test/spec/surface.test.js",
    },
    "system": Object {
      "hostname": "127.0.0.1",
      "pid": 36255,
      "process_name": "/Users/thanpolas/.nvm/versions/node/v10.1.0/bin/node",
    },
  },
  "dt": "2018-05-18T16:25:57.815Z",
  "event": Object {},
  "level": "notice",
  "message": "hello world",
}
`;

exports[`API Surface Logging Function Levels Testing log level: warn 1`] = `
Object {
  "context": Object {
    "runtime": Object {
      "application": "testLogality",
      "file": "/test/spec/surface.test.js",
    },
    "source": Object {
      "file_name": "/test/spec/surface.test.js",
    },
    "system": Object {
      "hostname": "127.0.0.1",
      "pid": 36255,
      "process_name": "/Users/thanpolas/.nvm/versions/node/v10.1.0/bin/node",
    },
  },
  "dt": "2018-05-18T16:25:57.815Z",
  "event": Object {},
  "level": "warn",
  "message": "hello world",
}
`;
